app/web/db/user_state.py

```diff
@@ -21,14 +20,14 @@ class UserState:
     def __init__(self, db: Session, email: str):
         self.db = db
         self.email = email.lower()
+        self._permissions = {}
 
     @property
     def permissions(self) -> Dict[str, GroupInfo]:
         """
         Returns a dict of all group permissions and a special {"all": read/archive_url/archive_sheet} key
         """
-        if not hasattr(self, '_permissions'):
-            self._permissions = {}
+        if not self._permissions:
```

Comment:
https://www.geeksforgeeks.org/python-check-if-dictionary-is-empty/

Reply:
ok, I see it is defined in the readme.
TODO for me is standardize those definitions in the readme